omod/src/main/java/org/openmrs/module/reporting/web/reports/ReportHistoryController.java

```diff
@@ -187,7 +188,21 @@ public String openFromHistory(@RequestParam("uuid") String uuid, HttpServletResp
 				req.setRenderingMode(mode);
 			}
 		}
+
+		List<ReportRequest> requests = rs.getAllReportRequests();
+		
+		List <String> allLogs = new ArrayList<String>();
+		for(ReportRequest reqst: requests) {
+			try {
+			List<String> logs = rs.loadReportLog(reqst);
+			allLogs.addAll(logs);
+			}catch(NullPointerException e) {				
```

Comment:
Is there a better way to deal with the nullpointer problem rather than catching it?